Add support for non-free MapQuest API

pratheekrebala opened the pull request:
Small addition to create support for the non-free MapQuest API. It's mostly copied from the Mapbox provider with slight changes to fit the Mapquest API (documented [here](https://developer.mapquest.com/documentation/geocoding-api/))

KostyaEsmukov commented on geopy/geocoders/__init__.py:
```diff
@@ -169,6 +170,7 @@
     "tomtom": TomTom,
     "what3words": What3Words,
     "yandex": Yandex,
+    "mapquest": MapQuest
```
1. The list should be sorted
2. Please take a look at the comment on the 81st line: it's not the only place where a new geocoder should be added.

pratheekrebala replied:
I think I added the geocoder in all the places. Could you confirm?

KostyaEsmukov commented on geopy/geocoders/mapquest.py:
```diff
@@ -0,0 +1,188 @@
+        """
+        :param str api_key: The API key required by Mapbox to perform
+            geocoding requests. API keys are managed through Mapox's account
+            page (https://www.mapbox.com/account/access-tokens).
```
Shouldn't that be MapQuest instead of MapBox?

pratheekrebala replied:
This is correct. I will make another commit this evening to change all these references. Apologies about the sight.

pratheekrebala replied:
All the references to MapBox have been changed to MapQuest.

KostyaEsmukov commented:
> Small addition to create support for the non-free MapQuest API. It's mostly copied from the Mapbox provider with slight changes to fit the Mapquest API (documented [here](http://developers.google.com/maps/documentation/geocoding))

This sentence is very confusing:
1. Does MapQuest use Mapbox-alike API (like OpenMapQuest uses Nominatim)? Or did you mean that you just chose MapBox as a skeleton for creating a new geocoding class (completely unrelated to MapBox)?
2. The "here" documentation link leads to Google, looks like it should've been https://developer.mapquest.com/documentation/geocoding-api/

pratheekrebala commented:
> > Small addition to create support for the non-free MapQuest API. It's mostly copied from the Mapbox provider with slight changes to fit the Mapquest API (documented [here](http://developers.google.com/maps/documentation/geocoding))
> 
> This sentence is very confusing:
> 
>    1. Does MapQuest use Mapbox-alike API (like OpenMapQuest uses Nominatim)? Or did you mean that you just chose MapBox as a skeleton for creating a new geocoding class (completely unrelated to MapBox)?

Sorry about the confusion. Your interpretation is correct: I used the MapBox as a skeleton to generate the Mapquest API.

>    2. The "here" documentation link leads to Google, looks like it should've been https://developer.mapquest.com/documentation/geocoding-api/

That's also the correct link to the documentation. I've updated my initial PR to point to the correct link.

pratheekrebala commented:
I think I have fixed all the changes you have requested and added tests to `test/geocoders/mapquest.py`. I ensured that the tests were passing on my local machine using my MapQuest key.

Let me know if you need a key to add to the CI.

Thank you for taking the time to review this PR!
